packages/contracts/contracts/minter-suite/Minters/MinterSetPriceV5.sol

```diff
@@ -0,0 +1,400 @@
+
+    /// minter version for this minter
+    string public constant minterVersion = "v5.0.0";
+
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    /// contractAddress => projectId => base project config
+    mapping(address => mapping(uint256 => ProjectConfig)) public projectConfig;
+
+    // STATE VARIABLES FOR SplitFundsLib
+    // contractAddress => IsEngineCache
+    mapping(address => SplitFundsLib.IsEngineCache) private _isEngineCaches;
+
+    // STATE VARIABLES FOR MaxInvocationsLib
+    /// contractAddress => projectId => max invocations specific project config
+    mapping(address => mapping(uint256 => MaxInvocationsLib.MaxInvocationsProjectConfig))
+        public maxInvocationsProjectConfig;
+
+    function _onlyArtist(
+        uint256 _projectId,
+        address _coreContract
+    ) internal view {
+        require(
+            msg.sender ==
+                IGenArt721CoreContractV3_Base(_coreContract)
+                    .projectIdToArtistAddress(_projectId),
+            "Only Artist"
+        );
+    }
+
+    /**
+     * @notice Initializes contract to be a Filtered Minter for
+     * `_minterFilter` minter filter.
+     * @param _minterFilter Minter filter for which this will be a
+     * filtered minter.
+     */
+    constructor(address _minterFilter) ReentrancyGuard() {
+        minterFilterAddress = _minterFilter;
+        minterFilter = IMinterFilterV1(_minterFilter);
+    }
+
+    /**
+     * @notice Returns whether or not the provided address `_coreContract`
+     * is an Art Blocks Engine core contract. Caches the result for future access.
+     * @param _coreContract Address of the core contract to check.
+     */
+    function _isEngine(address _coreContract) internal returns (bool) {
+        SplitFundsLib.IsEngineCache storage isEngineCache = _isEngineCaches[
+            _coreContract
+        ];
+        if (isEngineCache.isCached) {
+            return isEngineCache.isEngine;
+        } else {
+            bool isEngine = SplitFundsLib.getV3CoreIsEngine(
+                _coreContract,
+                isEngineCache
+            );
+            return isEngine;
+        }
+    }
+
+    /**
+     * @notice Syncs local maximum invocations of project `_projectId` based on
+     * the value currently defined in the core contract.
+     * @param _coreContract Core contract address for the given project.
+     * @param _projectId Project ID to set the maximum invocations for.
+     * @dev this enables gas reduction after maxInvocations have been reached -
+     * core contracts shall still enforce a maxInvocation check during mint.
+     */
+    function syncProjectMaxInvocationsToCore(
+        uint256 _projectId,
+        address _coreContract
+    ) public {
+        uint256 maxInvocations = MaxInvocationsLib
+            .syncProjectMaxInvocationsToCore(
+                _projectId,
+                _coreContract,
+                maxInvocationsProjectConfig
```

Comment:
nit: I think it would be a bit cleaner to pass in the `storage MaxInvocationsLib.MaxInvocationsProjectConfig` directly instead of the entire mapping. It removes the library's concern with how the minter decides to map to the structs, which I think makes the library a little more readable (and sometimes more efficient).
WDYT?

Reply:
ah yes I think I had some initial confusion/misunderstanding about applying storage updates to a struct that's nested inside a mapping, but yes there should be no issue/this makes a lot more sense to do :)